kvstore/badger/badger.go

```diff
@@ -148,29 +153,73 @@ func (s *badgerStore) DeletePrefix(prefix kvstore.KeyPrefix) error {
 
 func (s *badgerStore) Batched() kvstore.BatchedMutations {
 	return &batchedMutations{
-		batched:  s.instance.NewWriteBatch(),
-		dbPrefix: s.dbPrefix,
+		store:            s.instance,
+		dbPrefix:         s.dbPrefix,
+		setOperations:    make(map[string]kvstore.Value),
+		deleteOperations: make(map[string]types.Empty),
 	}
 }
 
 // batchedMutations is a wrapper around a WriteBatch of a BadgerDB.
 type batchedMutations struct {
-	batched  *badger.WriteBatch
-	dbPrefix []byte
+	store            *badger.DB
+	dbPrefix         []byte
+	setOperations    map[string]kvstore.Value
+	deleteOperations map[string]types.Empty
+	operationsMutex  sync.Mutex
 }
 
 func (b *batchedMutations) Set(key kvstore.Key, value kvstore.Value) error {
-	return b.batched.Set(append(b.dbPrefix, key...), value)
+	stringKey := typeutils.BytesToString(byteutils.Concat(b.dbPrefix, key))
```

Comment:
I get the feeling my comments have been lost :stuck_out_tongue: 

Instead use:
```
var b strings.Builder
b.Write(b.dbPrefix)
b.Write(key)
b.String()
```
This has the exact same effect, but its an stdlib function and not unsafe.

It could make sense to do `b.Grow(len(b.dbPrefix) + len(key)` as the lengths are known.